### PR TITLE
💅 Align heading typography to design

### PR DESCRIPTION
**Pull request by MeganSteenkamp**
* Update headings to use Centrapay typography.
* Add prose class at root to remove repetition.

Test plan: Expect Nuxt pages to have:
* Divider above `<h2>` headings
* Links in paragraphs to be underlines
* Headings to not be underlined

### CODE REVIEW

File: components/Card.vue [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+    class="not-prose card w-full p-2 rounded-xl shadow-sm border border-color-outline-transparent cursor-pointer hover:bg-interactive-tertiary-hover hover:shadow-md focus:ring-1 focus:ring-offset-2 focus:ring-focus-ring"
     :to="props.link"
   >
     <div class="p-2 space-y-2">
       <component
         :is="currentIcon"
         class="w-6 h-6 fill-brand-accent stroke-brand-accent"
       />
-      <h2>
+      <h2 class="type-headline-4">
```
Comment (tom-blake) [inline components/Card.vue:11]: @MeganSteenkamp I'm unfamiliar with these - where do these `type-*` styles come from?

File: app.vue [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+    prose-h3:type-headline-3 prose-h3:pt-8 prose-h3:scroll-mt-24 prose-h3:lg:scroll-mt-16
+    "
+  >
     <Header />
     <NuxtPage />
   </div>
 </template>
+
+<style scoped>
+.prose :where(h2 > a, h3 > a) {
```
Comment (MeganSteenkamp) [inline app.vue:15]: This isn't ideal **but** our the markdown content design has some links underlined (paragraphs) and some not (headings).

File: components/Card.vue (hunk 1 above)
Comment (MeganSteenkamp) [reply components/Card.vue:11]: @tom-blake these are defined in the `typography.js` file - they align with the class names use in Figma.

File: pages/index.vue [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <template>
   <main>
-    <div class="prose desktop-gutters px-4 sm:px-6">
-      <div class="my-4">
+    <div class="desktop-gutters px-4 sm:px-6">
+      <div class="not-prose my-4">
```
Comment (MeganSteenkamp) [inline pages/index.vue:4]: The prose classes take precedence to inline HTML classes so this class is required when there is an exception to the styling.

File: app.vue [hunk 4]
```diff
@@ -1,6 +1,20 @@
 <template>
-  <div>
+  <div
+    class="prose max-w-none
```
Comment (AndyClifford) [inline app.vue:3]: What is the purpose of these `prose` classes?

Comment (MeganSteenkamp) [reply app.vue:3]: @AndyClifford this indicates to use the Tailwind typography plugin. 
If you remove this locally you'll be able to see the difference easily 👍 
https://tailwindcss.com/docs/typography-plugin

File: app.vue [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
+    prose-h2:border-t prose-h2:pt-12 prose-h2:scroll-mt-20 prose-h2:lg:scroll-mt-12
```
Comment (MeganSteenkamp) [inline app.vue:4]: I initially had classes like `prose-h1:type-headline-1` here as this aligns with what is in Figma. I did a review with Junjie and we agreed the text styling given to us by the Tailwind typography plugin is preferred, therefore they have been removed 💅

File: components/Header.vue [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-    <div class="prose-a:no-underline max-w-7xl mx-auto flex items-center justify-start space-x-2">
+  <header class="not-prose w-full p-4 shadow-md sticky top-0 bg-surface-primary z-10">
+    <div class="max-w-7xl mx-auto flex items-center justify-start space-x-2">
       <NuxtLink
         to="/"
         class="flex items-center"
       >
         <icons-centrapay-logo class="icon-3xl fill-brand-accent" />
-        <h1 class="hidden md:block type-headline-4 px-1">
+        <span class="hidden md:block type-headline-4 px-1">
```
Comment (MeganSteenkamp) [inline components/Header.vue:9]: I had another look at this after I said it should be a `h1` - I think that feedback was incorrect sorry @Wulingoh <3
There should only be one h1 per page which will be in the content.

**Review by tom-blake — APPROVED** [review]

**Pull request merged**